src/libraries/System.Data.Common/src/System/Data/DataView.cs

```diff
@@ -1495,6 +1509,8 @@ internal void ResetRowViewCache()
             _rowViewCache = rvc;
         }
 
+        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026:UnrecognizedReflectionPattern",
+            Justification = "Row filter is already marked as unsafe.")]
```

Comment:
If it is just RowFilter that is unsafe, can we extract the code that is unsafe into its own method, so it limits the amount of code we are suppressing IL2026 for? That way we don't suppress too much.

Reply:
I see it is also creating a new `DataExpression`, which is RUC. So I kind of think this method should be marked `RUC` as well...

Reply:
What I meant here is that only DataExpression ctor is unsafe but user has already got that warning when they were setting RowFilter in the first place. If we annotate this as RUC then whole bunch of code will need to be RUC as well but user has already seen a warning and it will also flag many thing needlessly. I will extract it to new method to make it clearer

Reply:
note that setting RowFilter itself is not unsafe (it's just a string assignment) but it's marked as RUC because it's always used unsafely in DataExpression